certbot-nginx/certbot_nginx/nginxparser.py

```diff
@@ -1,23 +1,29 @@
 """Very low-level nginx config parser based on pyparsing."""
+import copy
+import logging
 import string
 
 from pyparsing import (
-    Literal, White, Word, alphanums, CharsNotIn, Forward, Group,
+    Literal, White, Word, alphanums, CharsNotIn, Combine, Forward, Group,
     Optional, OneOrMore, Regex, ZeroOrMore)
 from pyparsing import stringEnd
 from pyparsing import restOfLine
 
+logger = logging.getLogger(__name__)
 
 class RawNginxParser(object):
     # pylint: disable=expression-not-assigned
     """A class that parses nginx configuration with pyparsing."""
 
     # constants
+    space = Optional(White())
```

Comment:
Not to my knowledge, no.  The nginx grammar does not appear to be formally documented anywhere, and the canonical implementation is a hand-rolled parser in C.  I guess if we'd taken @olabini 's approach we could possibly have walked through and compared our hand-rolled parser with nginx's.